eisbuk-admin/src/components/temp/NewSlotForm/utils.ts

```diff
@@ -0,0 +1,32 @@
+import { DateTime } from "luxon";
+
+import { SlotFormValues } from "@/lib/data";
+
+import { SlotInterface } from "@/types/temp";
+
+import { fb2Luxon } from "@/utils/date";
+
+/**
+ * Process slot interface into form friendly values
+ * @param slot
+ * @returns
+ */
+export const slotToFormValues = (
+  slot: SlotInterface | undefined
+): (SlotFormValues & { date: DateTime }) | undefined => {
+  // fail early if no slot provided
+  if (!slot) return undefined;
+
+  const {
+    date: slotDate,
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
```

Comment:
Shouldn't we remove `id` from this assignment (since we actually don't use it) instead of disabling the linting warning?

Reply:
I think I understand now. The `id` variable will be needed later on when coding the full functionality, so for now we like to keep it here.

Reply:
No, to write it down (and make more sense then at the meeting): 

```typescript
interface InStruct {
  id: string;
  name: string;
  age: number;
}

interface OutStruct {
  name: string;
  age: number;
}

const ivan = {
  id: "13",
  name: "Ivan",
  age: 27,
};

const getPresonalInfo = (user: InStruct): OutStruct => {
  // split `id` of the user (type InStruct) and save the rest of the user as `presonalInfo`
  const { id, ...presonalInfo } = user;
  // return presonalInfo (type OutScruct)
  return presonalInfo;
};

const personalInfo = getPresonalInfo(ivan) // returns { name: "Ivan", age" 27 }
```

So here I'm using `id` declaration to put away the stuff I don't need from the object and return the rest (and the `id` is function scoped, so it gets garbage collected as soon as the function is done executing. This is just a syntactic sugar to not have to write it like this:

```typescript
const getPresonalInfo = (user: InStruct): OutStruct => ({
  name: user.name,
  age: user.age,
  // and you can imagine how many unnecessary lines
  // of code you get as the input object gets bigger
});
```

Reply:
Totally makes sense. Thanks for the detailed explanation!